File: teck2/OOP_trantor_2018/map.java
public class Map {
    private int x;
    private int y;
    private int rsc[];

    public static void map_check(Map map, Player ply) {
        for (int y = 0; y < rsc[7]; y ++)
            ply.setRep("Player ");
        for (int y = 0; y < rsc[0]; y ++)
            ply.setRep("food ");
        for (int y = 0; y < rsc[1]; y ++)
            ply.setRep("sibur ");
        for (int y = 0; y < rsc[2]; y ++)
            ply.setRep("phiras ");
        for (int y = 0; y < rsc[3]; y ++)
            ply.setRep("linemate ");
        for (int y = 0; y < rsc[4]; y ++)
            ply.setRep("mendiane ");
        for (int y = 0; y < rsc[5]; y ++)
            ply.setRep("thystame ");
        for (int y = 0; y < rsc[6]; y ++)
            ply.setRep("deraumere ");
    }

    protected void map_create(Server s) {
        Map cell = new Map();

        map_ srand(time(NULL));
        for (int y = 0; y < s.height; y++) {
            for (int x = 0; x < s.width; x++) {
                map_cell_fill(cell, y, x);
                s.m_list[y][x] = cell;
            }
        }
        s.m_list[s.height] = NULL;
    }

    protected void map_cell_fill(Map cell, int y, int x) {
        cell.x = x;
        cell.y = y;
        cell.rsc[0] = map_item_gen(40, 1, 3);
        cell.rsc[1] = map_item_gen(80, 1, 3);
        cell.rsc[2] = map_item_gen(60, 1, 3);
        cell.rsc[3] = map_item_gen(30, 1, 3);
        cell.rsc[4] = map_item_gen(30, 1, 3);
        cell.rsc[5] = map_item_gen(50, 1, 3);
        cell.rsc[6] = map_item_gen(10, 1, 2);
        cell.rsc[7] = 0;
    }
protected void map_create(Server s) {
	Map cell = new Map();

	srand(time(NULL));
	for (int y = 0; y < s.height; y++) {
		for (int x = 0; x < s.width; x++) {
			map_cell_fill(cell, y, x);
			s.m_list[y][x] = cell;
		}
	}
	s.m_list[s.height] = NULL;
}

protected void map_cell_fill(Map cell, int y, int x) {
	cell.x = x;
	cell.y = y;
	cell.rsc[0] = map_item_gen(40, 1, 3);
	cell.rsc[1] = map_item_gen(80, 1, 3);
	cell.rsc[2] = map_item_gen(60, 1, 3);
	cell.rsc[3] = map_item_gen(30, 1, 3);
	cell.rsc[4] = map_item_gen(30, 1, 3);
	cell.rsc[5] = map_item_gen(50, 1, 3);
	cell.rsc[6] = map_item_gen(10, 1, 2);
	cell.rsc[7] = 0;
}

protected String map_cell_check(Server s, Map m, String str)
{
    for (int x = 0; x < m.rsc[7]; x++) {
        str = string_concat(str, s.item[7]);
        str = string_concat(str, " ");
    }
    for (int y = 0; y < 7; y++) {
        for (int x = 0; x < m.rsc[y]; x++) {
            str = string_concat(str, s.item[y]);
            str = string_concat(str, " ");
        }
    }
    str[string_lenght(str) - 1] = '\0';
    return (str);
}

protected int map_item_gen(int percent, int min, int max)
{
    int rslt = map_rand() % 100 + 1;

    if (rslt <= percent)
        return (map_rand() % (max - min) + min);
    return (0);
}

    protected String map_cell_check(Server s, Map m, String str)
    {
        for (int x = 0; x < m.rsc[7]; x++) {
            str = string_concat(str, s.item[7]);
            str = string_concat(str, " ");
        }
        for (int y = 0; y < 7; y++) {
            for (int x = 0; x < m.rsc[y]; x++) {
                str = string_concat(str, s.item[y]);
                str = string_concat(str, " ");
            }
        }
        str[string_lenght(str) - 1] = '\0';
        return (str);
    }

    protected int map_item_gen(int percent, int min, int max)
    {
        int rslt = map_rand() % 100 + 1;

        if (rslt <= percent)
            return (map_rand() % (max - min) + min);
        return (0);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int[] getRsc() {
        return rsc;
    }
}

File: teck2/USP_cloning_trantor_2018/map.c
/*
** EPITECH PROJECT, 2018
** map.c
** File description:
** map functions
*/

#include "./include/cloning_trantor.h"

void map_set(srv *srv)
{
    srv->m_list = malloc(sizeof(cell*) * srv->height);
    cell c;

    srand(time(NULL));
    for (int i = 0; i < srv->height; i++) {
        srv->m_list[i] = malloc(sizeof(cell) * srv->width);
        for (int j = 0; j < srv->width; j++) {
            map_fill(&c, i, j);
            srv->m_list[i][j] = c;
        }
    }
    srv->m_list[srv->height] = NULL;
}

void map_fill(cell *c, int i, int j)
{
    c->y = i;
    c->x = j;
    c->rsc = malloc(sizeof(int) * 8);
    c->rsc[0] = rand() % 3;
    c->rsc[1] = rand() % 5;
    c->rsc[2] = rand() % 3;
    c->rsc[3] = rand() % 3;
    c->rsc[4] = rand() % 3;
    c->rsc[5] = rand() % 3;
    c->rsc[6] = rand() % 2;
    c->rsc[7] = 0;
}

char *map_check(srv *srv, cell m, char *src)
{
    for (int i = 0; i < m.rsc[7]; i++) {
        src = string_concat(src, srv->items[7]);
        src = string_concat(src, " ");
    }
    for (int i = 0; i < 7; i++) {
        for (int j = 0; j < m.rsc[i]; j++) {
            src = string_concat(src, srv->items[i]);
            src = string_concat(src, " ");
        }
    }
    src[string_lenght(src) - 1] = '\0';
    return (src);
}
